### PR TITLE
Update stableswap solver to use new math and convert it to `BigDec`

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
This PR updates our stableswap solver to use the most up-to-date math described in our spec. It also converts all core logic to use our custom decimal type implemented in #1021 while leaving the external-facing interface still on `Dec`

It also fixes the solver tests to so they ensure that k increases with any core operation (i.e. any slippage/rounding error is in the pool's favor) instead of passing trivially

### CODE REVIEW

File: packages/math/src/pool/stable.ts [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
+import { BigDec } from "../big-dec";
 
 export const StableSwapMath = {
   calcOutGivenIn,
   calcInGivenOut,
   calcSpotPrice,
   // secondary
   solveCfmm,
-  compare_checkMultErrorTolerance,
+  compareDec_checkMultErrorTolerance,
```
Comment (jonator) [inline packages/math/src/pool/stable.ts:10]: We should only add these if they need to be exposed to users of the package (the web package, currently), not just tests

**Comment by jonator** [comment]
Would I just merge #1022 and not #1021 ?

File: packages/math/src/pool/stable.ts [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
-const oneDec = new Dec(1);
+const oneBigDec = new BigDec(1);
 
 export type StableSwapToken = {
   amount: Dec;
   denom: string;
   scalingFactor: number;
 };
 
+export type BigDecStableSwapToken = {
```
Comment (jonator) [inline packages/math/src/pool/stable.ts:23]: Remove export

**Comment by AlpinYukseloglu** [comment]
> Would I just merge #1022 and not #1021 ?

That works!

**Pull request merged**